esphome/components/pmsx003/pmsx003.cpp

```diff
@@ -173,163 +175,143 @@ optional<bool> PMSX003Component::check_byte_() {
   return {};
 }
 
-void PMSX003Component::send_command_(uint8_t cmd, uint16_t data) {
-  this->data_index_ = 0;
-  this->data_[data_index_++] = 0x42;
-  this->data_[data_index_++] = 0x4D;
-  this->data_[data_index_++] = cmd;
-  this->data_[data_index_++] = (data >> 8) & 0xFF;
-  this->data_[data_index_++] = (data >> 0) & 0xFF;
-  int sum = 0;
-  for (int i = 0; i < data_index_; i++) {
-    sum += this->data_[i];
+bool PMSX003Component::check_payload_length_(uint16_t payload_length) {
+  switch (this->type_) {
+    case PMSX003_TYPE_X003:
+      return payload_length == 28 || payload_length == 20;  // 2*13+2 // TODO: payload_length=20?
```

Comment:
Could you clarify where you saw the TODO comment? Did you forgot to remove it?

Reply:
the check for `payload_length == 20` was already there (https://github.com/esphome/esphome/pull/8344/files#diff-999cd0d338a825182fde2018b73bd2e625969160484c41e9bcdb012147a7a88fL138) but i couldn't find any documentation for that case. so i'm not sure if this was wrong in the past. Any suggestions for a better comment? like: "No documentation found for `payload_length == 20`. Keeping it to don't break anything"

Reply:
```suggestion
      // The expected payload length is typically 28 bytes.
      // However, a 20-byte payload check was already present in the code.
      // No official documentation was found confirming this.
      // Retaining this check to avoid breaking existing behavior.
      return payload_length == 28 || payload_length == 20;
```

Maybe this is enough?